### PR TITLE
Update rio-tiler to v4.1.7

### PR DESCRIPTION
**Pull request by giswqs**
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by vincentsarago** [comment]
@conda-forge-admin, please re-render

**Comment by giswqs** [comment]
If I recalled correctly, it was the color-operations package that's not available on conda-forge.

https://github.com/cogeotiff/rio-tiler/blob/5e5e33e8fd8a744276c698c4400f9416eb932d66/setup.py#LL37C10-L37C26

**Comment by vincentsarago** [comment]
yeah it's only available on pypi

**Comment by giswqs** [comment]
Let me submit a conda-forge recipe and add you as a maintainer?

File: recipe/meta.yaml [hunk 1]
```diff
@@ -17,22 +18,23 @@ build:
...
     - numexpr
     - numpy
+    - morecantile >=3.1,<4.0
     - pydantic
     - pystac >=0.5.4
-    - python >=3.7,<3.9
-    - rasterio >=1.1.7
-    - rio-color
+    - rasterio >=1.3.0
+    # - color-operations
```
Comment (vincentsarago) [inline recipe/meta.yaml:36]:
```suggestion
    - pip:
      - color-operations
```

Comment (giswqs) [reply recipe/meta.yaml:36]: conda-forge does not using pip in the recipe. We better get `color-operations` on conda-forge.

**Comment by vincentsarago** [comment]
> add you as a maintainer?

@giswqs thanks but I would prefer not to 🙈. I'm not a big fan of conda and I don't want to add more maintenance work on my plate. I just came here because one of my colleague was asking about rio-tiler on conda

File: recipe/meta.yaml (hunk 1 above)
Comment (vincentsarago) [reply recipe/meta.yaml:36]: > We better get color-operations on conda-forge.

up to you 👍

**Comment by giswqs** [comment]
Adding color-operations: https://github.com/conda-forge/staged-recipes/pull/22474

**Comment by jsignell** [comment]
How are you building this locally @giswqs? I tried running `python build-locally.py` but I am getting a different error.

```python-traceback
+ echo 'Finding recipes merged in main and removing them from the build.'
Finding recipes merged in main and removing them from the build.
+ pushd /home/conda/staged-recipes/recipes
+ '[' False == True ']'
+ git ls-tree --name-only main -- .
+ xargs -I '{}' sh -c 'rm -rf ~/staged-recipes-copy/recipes/{} && echo Removing recipe: {}'
fatal: detected dubious ownership in repository at '/home/conda/staged-recipes'
To add an exception for this directory, call:

	git config --global --add safe.directory /home/conda/staged-recipes
Traceback (most recent call last):
  File "/home/jsignell/staged-recipes/build-locally.py", line 101, in <module>
    main()
  File "/home/jsignell/staged-recipes/build-locally.py", line 95, in main
    run_docker_build(ns)
  File "/home/jsignell/staged-recipes/build-locally.py", line 33, in run_docker_build
    subprocess.check_call([script])
  File "/home/jsignell/conda/envs/build/lib/python3.10/subprocess.py", line 369, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['.scripts/run_docker_build.sh']' returned non-zero exit status 1.
```

**Comment by jsignell** [comment]
I just stumbled across these docs (https://conda-forge.org/docs/maintainer/knowledge_base.html#building-against-numpy) which seem to indicate that maybe numpy needs some special treatment? I'll give it a shot.

**Comment by jsignell** [comment]
I'm testing that theory out on https://github.com/conda-forge/staged-recipes/pull/22480

**Comment by giswqs** [comment]
@jsignell Thanks for looking into it. I just added `{{ pin_compatible('numpy') }}` to run.

https://github.com/conda-forge/staged-recipes/pull/22480/files#diff-5abf3216d5d6b92da3ab0f19fb1adb91792de13eae82969ca3a913e6637bdb19R27

**Comment by giswqs** [comment]
Still the same error

> ModuleNotFoundError: No module named 'color_operations.colorspace'

**Comment by giswqs** [comment]
The build went smoothly. The issue is the `import color_operations` check that fails.

**Comment by jsignell** [comment]
yeah did you say you were able to build the package locally?

**Comment by giswqs** [comment]
No, building locally uses the same docker image, and I am pretty sure it will throw the same errors. What I meant run fine locally is that I installed the package in a fresh conda env with Python 3.11 (the same version of the conda-forge recipe), and no errors when `import color_operations`

**Comment by giswqs** [comment]
I think the problem is building this c file. 
https://github.com/vincentsarago/color-operations/blob/main/color_operations/colorspace.c

**Comment by vincentsarago** [comment]
oh interesting I think this file shouldn't be there 😓

File: recipe/meta.yaml
Comment (jsignell) [inline recipe/meta.yaml:1]: I would be happy being added as a maintainer of this package and I think you should consider adding yourself as well @giswqs.

Comment (giswqs) [reply recipe/meta.yaml:1]: You can add user and auto-merge bot using the command when the recipe is accepted by conda-forge. 
https://conda-forge.org/docs/maintainer/infrastructure.html#admin-web-services

Comment (jsignell) [reply recipe/meta.yaml:1]: oooh nice!

**Comment by jsignell** [comment]
This should be ready to go now that color-operations is on conda-forge.

**Comment by jsignell** [comment]
@conda-forge-admin, please restart ci

**Review by jsignell — APPROVED** [review]
LGTM!!!

**Comment by jsignell** [comment]
@conda-forge-admin, please add user @jsignell

**Comment by jsignell** [comment]
@lsetiawan, @nrweir this is ready for your approval and merge.

**Comment by dopplershift** [comment]
@jsignell The command to add a user (as noted in the docs) should be in the *title* of a new issue. We've also been having some issues, so don't put a comma after "conda-forge-admin". I'll be happy to merge the PR the bot creates.

**Comment by jsignell** [comment]
Thanks Ryan! That worked perfectly :) Would you be able to merge this PR as well?

**Comment by dopplershift** [comment]
Once you accept the invite for the repo, you can merge this PR yourself.

**Pull request merged**